team2_pii.py

```diff
@@ -21,26 +26,81 @@ def find_credit_card_number(text):
     if match:
         return True
     return False
+
+def is_leap_year(year):
+    if year % 4 == 0:
+        if year % 100 == 0 and year % 400 != 0:
+            return False
+        return True
+    return False
+
+def find_dob(text):
+    match = re.search(r'(\d{1,2})[-|\/](\d{1,2})[-|\/]((\d{4})|(\d{2}))', text)
+    if match:
+        dob_lst = match.groups()
+        month = int(dob_lst[0])
+        day = int(dob_lst[1])
+        year = int(dob_lst[2])
+
+        if (month < 1 or month > 12) or (day < 1 or day > 31):
+            return False
+        elif (month == 2 and (day > 29 or (day == 29 and not is_leap_year(year)))):
+            return False
+        else:
+            return True
+    return False
+
+def find_name(text):
```

Comment:
You made name detection significantly more complicated than necessary for this assignment.  I would suggest looking names such as Jon Smith by finding a capital letter [A-Z] followed by one or more letters [a-z]+ then a space \s and then repeat the pattern.  You'll find this reduces your exclude list significantly.  
```
[A-Z][a-z]+\s[A-Z][a-z]+
```
Trying to find names which are not capitalized such as "jane doe" is very difficult to accomplish without using some form of natural language processing.  If you are interested in that I can definitely send some examples, however, it wasn't required for this assignment. 

The problem with your current code is that it will find multiple things which are not names, for example, here is a test case which should pass but does not:
```
        name = 'SSN: 123-456-7890'
        expected = [[], ['SSN']]
        self.assertEqual(find_name(name), expected)
```